### ISSUE TITLE
RE: new project yarn test failed (error message:This promise must be present when running with -o.)

### ISSUE THREAD

**Issue by 75341704** [facebook/react-native#17562]
### Is this a bug report?
Maybe.

### Environment
Environment:
  OS: macOS High Sierra 10.13.2
  Node: 6.9.1
  Yarn: 1.3.2
  npm: 4.1.1
  Watchman: 4.9.0
  Xcode: Xcode 9.2 Build version 9C40b
  Android Studio: 2.2 AI-145.3537739

Packages: (wanted => installed)
  react: 16.0.0 => 16.0.0
  react-native: 0.50.3 => 0.50.3

### Steps to Reproduce

1. create-react-native-app foo
2.cd foo
3.yarn test

### Expected Behavior
The default App.test.js is running and show all passed.

### Actual Behavior
error message:

Determining test suites to run...Error: This promise must be present when running with -o.
    at /Users/developer/Desktop/code/es6/rn/learnTest/node_modules/jest-cli/build/search_source.js:202:17

**Comment by 75341704** [comment]
But If it is react project, steps below
1.create-react-app foo
2.cd foo
3.yarn test

The default App.test.js is working and show all passed.

Anyone have some idea abt it?

**Comment by jcollum** [comment]
Same here. Create a brand new project, test fails to run. 

```Determining test suites to run...Error: This promise must be present when running with -o.
    at /Users/justin/projects/CadmiumCardinal/node_modules/jest-cli/build/search_source.js:202:17
    at Generator.next (<anonymous>)
    at step (/Users/justin/projects/CadmiumCardinal/node_modules/jest-cli/build/search_source.js:19:362)
    at /Users/justin/projects/CadmiumCardinal/node_modules/jest-cli/build/search_source.js:19:592
    at Promise (<anonymous>)
    at /Users/justin/projects/CadmiumCardinal/node_modules/jest-cli/build/search_source.js:19:273
    at SearchSource.getTestPaths (/Users/justin/projects/CadmiumCardinal/node_modules/jest-cli/build/search_source.js:216:10)
    at /Users/justin/projects/CadmiumCardinal/node_modules/jest-cli/build/run_jest.js:44:29
    at Generator.next (<anonymous>)
    at step (/Users/justin/projects/CadmiumCardinal/node_modules/jest-cli/build/run_jest.js:23:380)

```

**Comment by 75341704** [comment]
@jcollum The --watch argument is deleted in the package.json 
=QTE=
"test": "node node_modules/jest/bin/jest.js --watch"
=UQTE=
The yarn test is working but It fail to watch the files' change.

**Comment by jcollum** [comment]
I don't understand what you are trying to say. The `--watch` argument is present in the package.json: 

```
$ cat package.json | grep watch
    "test": "node node_modules/jest/bin/jest.js --watch"
```

This is from a fresh app, I have not modified it. The problem appears to be with the `--watch` option: 

```
$ node node_modules/jest/bin/jest.js
 PASS  ./App.test.js
  ✓ renders without crashing (2894ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        6.005s
Ran all test suites.
```

versus: 

```
$ npm test 

[screen clears]

Determining test suites to run...Error: This promise must be present when running with -o.
    at /Users/justin/projects/CadmiumCardinal/node_modules/jest-cli/build/search_source.js:202:17
    at Generator.next (<anonymous>)
    at step (/Users/justin/projects/CadmiumCardinal/node_modules/jest-cli/build/search_source.js:19:362)
    at /Users/justin/projects/CadmiumCardinal/node_modules/jest-cli/build/search_source.js:19:592
    at Promise (<anonymous>)
    at /Users/justin/projects/CadmiumCardinal/node_modules/jest-cli/build/search_source.js:19:273
    at SearchSource.getTestPaths (/Users/justin/projects/CadmiumCardinal/node_modules/jest-cli/build/search_source.js:216:10)
    at /Users/justin/projects/CadmiumCardinal/node_modules/jest-cli/build/run_jest.js:44:29
    at Generator.next (<anonymous>)
    at step (/Users/justin/projects/CadmiumCardinal/node_modules/jest-cli/build/run_jest.js:23:380)
```

**Comment by 75341704** [comment]
@jcollum 
Firstly, if you want to make the 'npm test' command success, the --watch should be deleted  in package.json file 
...
 "test": "node node_modules/jest/bin/jest.js "
...

Secondly, "--watch" in the package.json, I think, make "npm test " fail.

Last but not least, the --watch is deleted and then you have to run 'npm test' command whenever the files are changed.

**Comment by jcollum** [comment]
>the --watch should be deleted in package.json file

Right, it appears to be a bug, which is why I came here to file an issue.

**Comment by jcollum** [comment]
This appears to have been fixed in 0.53.3

```
react-native init testBug
...
cd testBug/
npm test

> testBug@0.0.1 test /Users/justin/projects/testBug
> jest

 PASS  __tests__/App.js (5.853s)
  ✓ renders correctly (3691ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        6.87s
Ran all test suites.


```
